AppServer/google/appengine/tools/devappserver2/wsgi_request_info.py

```diff
@@ -159,3 +159,16 @@ def get_instance(self, request_id):
     """
     with self._lock:
       return self._request_id_to_instance.get(request_id, None)
+
+  def get_scheme(self, request_id):
+    """Returns the scheme for this request.
+
+    Args:
+      request_id: The string id of the request making the API call.
+
+    Returns:
+      One of 'http', 'https'.
+    """
+    with self._lock:
+      environ = self._request_wsgi_environ[request_id]
+      return environ['HTTP_X_FORWARDED_PROTO']
```

Comment:
I would suggest setting it to http (or https) and overriding it if the header indicates otherwise. If for some reason a request comes in without that header, we still want to return one of the two values.